optuna/cli.py

```diff
@@ -228,7 +231,8 @@ def _dump_table(records: list[dict[str, Any]], header: list[str]) -> str:
     ret += separator + "\n"
     ret += header_string + "\n"
     ret += separator + "\n"
-    ret += "\n".join(rows_string) + "\n"
+    if rows:
+        ret += "\n".join(rows_string) + "\n"
```

Comment:
```suggestion
    for row_string in rows_string:
        ret += row_string + "\n"
```
It might be more straightforward to simply iterate over rows_string.